mataa_tools/mataa_ESS_sync.m

```diff
@@ -0,0 +1,117 @@
+% > mataa_plot_IR(h, t);																						% plot the IR
+% > set(findall(gcf, 'type', 'axes'), 'xgrid', 'on', 'xminorgrid', 'on', 'ygrid', 'on', 'yminorgrid', 'on');	% make it easier to see the coordinates of a part of the response
+% > t0 = mataa_guess_IR_start(h, fs);																			% find the beginning of the linear response
+% > xlim([t0 t0+0.01]);																							% focus on the first 10 ms of the linear response
+% > tfirstecho = 5.0511-t0;																						% time of first reflexion, as seen in the plot, relative to beginning of response
+% > tharm = round(fs * T) .* log([2:5]) ./ (fs * log(2 ^ P));													% beginning of higher order responses (up to order 5) relative to beginning of linear response
+% > tharm2 = t0-tharm(1);																						% beginning of second order response
+% > xlim([tharm2 tharm2+tfirstecho]);																			% focus on the part of the plot corresponding to the second order response
+% > ylim([-0.001 0.001]);																						% change scale to be able to see a small second order response
+% > hc = mataa_signal_crop(h, fs, t0, t0+tfirstecho);															% extract the linear response
+% > [mag, phase, f] = mataa_IR_to_FR(hc, fs);																	% get the linear frequency response
+% > hc2 = mataa_signal_crop(h, fs, tharm2, tharm2+tfirstecho);													% extract the second order response
+% > [mag2, phase2, f2] = mataa_IR_to_FR(hc2, fs);																% get the second order frequency response
+% > semilogx(f, mag, ';Fundamental;');																			% plot the linear frequency response magnitude
+% > hold on;																									% let other curve be on the same plot
+% > semilogx(f2, mag2, ';2nd harmonic;');																		% plot the second order frequency response magnitude
+% 
+% DISCLAIMER:
+% This file is part of MATAA.
+% 
+% MATAA is free software; you can redistribute it and/or modify
+% it under the terms of the GNU General Public License as published by
+% the Free Software Foundation; either version 2 of the License, or
+% (at your option) any later version.
+% 
+% MATAA is distributed in the hope that it will be useful,
+% but WITHOUT ANY WARRANTY; without even the implied warranty of
+% MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+% GNU General Public License for more details.
+% 
+% You should have received a copy of the GNU General Public License
+% along with MATAA; if not, write to the Free Software
+% Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+% 
+% Copyright (C) 2006, 2007, 2008,2015 Matthias S. Brennwald.
+% Contact: info@audioroot.net
+% Further information: http://www.audioroot.net/MATAA
+
+if (P - fix(P)) ~= 0
+  error("P is not an integer\n");
+endif
+if P <= 0
+  error("P is not greater than 0\n");
+endif
+
+# Sweep Parameters
+N = round(fs * T);
+n = [0:N]';
+M = round(N / (log(2 ^ P) * (2 ^ (P + 1))));
+
+disp("Note that if later mataa_IR_to_FR is called, it adds 112 dB to the magnitude");
```

Comment:
I don't see the point of this message. I'd suggest to remove this to avoid confusion.